### PR TITLE
feature: included ERC20 contract address among the data in tokens.ckerc20.json

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

It will be useful to save the ERC20 contract address of the twin token during the job to create the JSON, for example for PR #1471 .

# Changes

- Deconstruct property `erc20_contract` and map it.
- Adjust zod type.

# Tests

Running locally, the script worked and there was no disruption in the replica.

### CODE REVIEW

File: scripts/build.tokens.ckerc20.mjs [hunk 1]
```diff
@@ -24,9 +24,10 @@ const orchestratorInfo = async ({ orchestratorId: canisterId }) => {
 const buildOrchestratorInfo = async (orchestratorId) => {
 	const { managed_canisters } = await orchestratorInfo({ orchestratorId });
 
-	const mapManagedCanisters = (acc, { ledger, index, ckerc20_token_symbol }) => {
+	const mapManagedCanisters = (acc, { ledger, index, ckerc20_token_symbol, erc20_contract }) => {
 		const ledgerCanister = fromNullable(ledger);
 		const indexCanister = fromNullable(index);
+		const erc20ContractAddress = erc20_contract.address;
```
Comment (peterpeterparker) [inline scripts/build.tokens.ckerc20.mjs:30]: wanna get fancy and spare this line? you can destruct the `address` when the param is destructed.

```
const mapManagedCanisters = (acc, { ledger, index, ckerc20_token_symbol, erc20_contract: {address} }) => {
```

File: src/frontend/src/icp/types/env-token-ckerc20.ts [hunk 2]
```diff
@@ -2,7 +2,8 @@ import { z } from 'zod';
 
 const envTokenData = z.object({
 	ledgerCanisterId: z.string(),
-	indexCanisterId: z.string()
+	indexCanisterId: z.string(),
+	erc20ContractAddress: z.string()
```
Comment (peterpeterparker) [inline src/frontend/src/icp/types/env-token-ckerc20.ts:6]: I was curious to see if there was a zod way to check if the address starts with `0x` and landed on this SO answer https://ethereum.stackexchange.com/a/146766

WDYT?

In our setup:

```
import { isEthAddress } from '$lib/utils/account.utils';
import { z } from 'zod';

z.custom<string>(isEthAddress, "Invalid Address")
```

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

Share two ideas

File: scripts/build.tokens.ckerc20.mjs (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply scripts/build.tokens.ckerc20.mjs:30]: yes!!!

File: src/frontend/src/icp/types/env-token-ckerc20.ts (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/types/env-token-ckerc20.ts:6]: i like it!

**Pull request merged**